torch/nn/utils/stateless.py

```diff
@@ -0,0 +1,121 @@
+import contextlib
+from typing import Any, Callable, Dict, Iterator, List, Tuple
+
+import torch
+from torch import Tensor
+
+__all__ = ["functional_call"]
+
+# We avoid typing module here because module attributes are declared as Union[Parameter, Tensor] by default
+# and using other types causes mypy errors
+def _change_class(module) -> None:
+    cls = module.__class__
+    func_params : Dict[str, Tensor] = module._functional_parameters
+
+    def _getattribute(self, name: str) -> Any:
+        if name in func_params:
+            return func_params[name]
+        return cls.__getattribute__(self, name)
+
+    param_cls = type(
+        f"StatelessReplacer{cls.__name__}",
+        (cls,),
+        {
+            "__getattribute__": _getattribute,
+        },
+    )
+
+    module.__class__ = param_cls
+    module._orig_class = cls
+
+
+def _swap_parameters(module, tensor_name: str, tensor: Tensor) -> None:
+    # Changes the module class to get a new __getattr__ dunder method
+    # that looks for the reparametrized tensor
+    if hasattr(module, "_functional_parameters"):
+        module._functional_parameters[tensor_name] = tensor
+    else:
+        module._functional_parameters = {}
+        module._functional_parameters[tensor_name] = tensor
+        _change_class(module)
+
+
+def _remove_swap(module, name: str) -> None:
+    if hasattr(module, "_orig_class"):
+        module.__class__ = module._orig_class
+        delattr(module, "_orig_class")
+        delattr(module, "_functional_parameters")
+
+
+@contextlib.contextmanager
+def _reparametrize_module(
+    module: 'torch.nn.Module',
+    parameters_and_buffers: Dict[str, Tensor],
+) -> Iterator[None]:
+    for name, tensor in parameters_and_buffers.items():
+        _apply_func_submodules(
+            _swap_parameters,
+            module, name.split("."), (tensor,))
+    yield
+    for name in parameters_and_buffers:
+        _apply_func_submodules(
+            _remove_swap,
+            module, name.split("."), ())
+
+
+def _apply_func_submodules(
+    func: Callable[..., None],
+    module: 'torch.nn.Module',
+    path: List[str],
+    args: Tuple,
+):
+    if len(path) == 1:
+        func(module, path[0], *args)
+    else:
+        _apply_func_submodules(func, getattr(module, path[0]), path[1:], args)
+
+
+def functional_call(
+    module: 'torch.nn.Module',
+    parameters_and_buffers: Dict[str, Tensor],
+    args: Tuple,
+    kwargs : Dict[str, Any] = None,
+):
+    r"""Performs a functional call on the module by replacing the module parameters
+    and buffers with the provided ones.
+
+    .. note:: If the module has active parametrizations, passing a value in the
+        `parameters_and_buffers` argument with the name set to the regular parameter
+         name will completely disable the parametrization.
```

Comment:
Do you want to add this to a .rst file so it shows up in our documentation?

Reply:
Yes, I actually wanted to make sure CI would fail as a test

Reply:
It did not :(
Should be fixed now so trying again.